include/deal.II/fe/fe_bernardi_raugel.h

```diff
@@ -0,0 +1,106 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2004 - 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_fe_bernardi_raugel_h
+#define dealii_fe_bernardi_raugel_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/geometry_info.h>
+#include <deal.II/base/polynomial.h>
+#include <deal.II/base/polynomials_bernardi_raugel.h>
+#include <deal.II/base/table.h>
+#include <deal.II/base/tensor_product_polynomials.h>
+
+#include <deal.II/fe/fe.h>
+#include <deal.II/fe/fe_poly_tensor.h>
+
+#include <vector>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * The Bernardi-Raugel element.
+ *
+ * <h3>Degrees of freedom</h3>
+ * The BR1 element has <i>dim</i> degrees of freedom on each node and 1 on each
+ * face. The shape functions are ordered by the $(Q_1)^d$ shape functions
+ * supported on each vertex, increasing according to vertex ordering on the
+ * element in GeometryInfo, then the bubble functions follow in the ordering
+ * given in PolynomialsBernardiRaugel.
+ *
+ * This element only has 1 degree (degree $p=1$) because it yields an LBB stable
+ * pair BR1-P0 for Stokes problems which is lower degree than the Taylor-Hood
+ * element. The pair is sometimes referred to as an enriched P1-P0 element or a
+ * reduced P2-P0 element.
+ *
+ * This element does not support hanging nodes or multigrid in the current
+ * implementation.
+ *
+ */
+template <int dim>
+class FE_BernardiRaugel
+  : public FE_PolyTensor<PolynomialsBernardiRaugel<dim>, dim>
+{
+public:
+  /**
+   * Constructor for the Bernardi-Raugel element of degree @p p. The only
+   * supported degree is 1.
+   *
+   * @arg p: The degree of the element $p=1$ for $BR_1$.
+   */
+  FE_BernardiRaugel(const unsigned int p = 1);
+
+  /**
+   * Return a string that uniquely identifies a finite element. This class
+   * returns <tt>FE_BR<dim>(degree)</tt>, with @p dim and @p degree replaced
+   * by appropriate values.
+   */
+  virtual std::string
+  get_name() const;
+
+  virtual std::unique_ptr<FiniteElement<dim, dim>>
+  clone() const;
```

Comment:
Please mark these virtual functions as `override`.

Reply:
I believe I got them. I'm not strong on my virtual functions so let me know if I messed up the syntax.

Reply:
Looks good to me.